File: 2025 APRIL(LEETCODE)/DAY 5/DAY_5.cpp
// 1863. Sum of All Subset XOR Totals
#include<iostream>
#include<vector>

using namespace std;
class Solution
{
public:
    int subsetXORSum(vector<int> &nums)
    {
        int ans = 0;
        for (int num : nums)
        {
            ans |= num;
        }
        return ans << (nums.size() - 1);
    }
};

File: 2025 APRIL(LEETCODE)/DAY 5/DAY_5.java
// 1863. Sum of All Subset XOR Totals
public class DAY_5 {
    public int subsetXORSum(int[] nums) {
        int result = 0;
        for(int num : nums) {
            result |= num;
        }
        return result << (nums.length - 1);
    }
}
